app/models/route_set.rb

```diff
@@ -67,7 +67,9 @@ def register!
         register_gone_route(route)
       end
     else
-      register_rendering_app
+      if SET_ROUTER_BACKEND_ADDRESS_ON_PUBLISH
+        register_rendering_app
+      end
```

Comment:
Arguable small stylistic improvement:

```suggestion
      register_rendering_app if SET_ROUTER_BACKEND_ADDRESS_ON_PUBLISH
```

you could also consider skipping the negation and going with

```suggestion
      register_rendering_app unless NO_SET_ROUTER_BACKEND_ADDRESS_ON_PUBLISH
```

which I think reads okay, but is probably personal taste